crates/subspace-service/src/dsn/piece_record_store.rs

```diff
@@ -82,21 +72,13 @@ where
             &[],
         )?;
 
-        // Remove obsolete pieces once in TOLERANCE_SEGMENTS_NUMBER times
-        let segment_index = first_piece_index / PIECES_IN_SEGMENT as u64;
-
-        let starting_piece_index = segment_index
-            .checked_sub(self.max_segments_number_in_cache() + TOLERANCE_SEGMENTS_NUMBER - 1)
-            .map(|starting_segment_index| starting_segment_index * PIECES_IN_SEGMENT as u64);
-
-        let pieces_to_delete_number =
-            (TOLERANCE_SEGMENTS_NUMBER * PIECES_IN_SEGMENT as u64) as usize;
-        if let Some(starting_piece_index) = starting_piece_index {
-            let keys = (starting_piece_index..)
-                .take(pieces_to_delete_number)
+        if let Some(delete_pieces_from_index) =
+            first_piece_index.checked_sub(self.max_pieces_in_cache)
+        {
+            let keys = (delete_pieces_from_index..first_piece_index)
+                .into_iter()
```

Comment:
Isn't it useless here?

Reply:
What exactly?

Reply:
`.into_iter()` method on the range. `Range` is an iterator already

Reply:
Not necessary indeed, will clean up. Sharp eye :wink: